projects/vdk-plugins/vdk-ipython/tests/test_plugin.py

```diff
@@ -169,3 +180,66 @@ def test_extension_with_pure_sql_job(ip, tmpdir):
     assert "2021-01-01  GOOG          123" in (
         ip.get_ipython().getoutput("! " "vdk " "sqlite-query -q 'SELECT * FROM stocks'")
     )
+
+
+def test_finalise(ip):
+    ip.run_line_magic(magic_name="reload_VDK", line="")
+
+    ip.get_ipython().run_cell("job_input = VDK.get_initialized_job_input()")
+
+    # check whether finalize_job is in the output before calling finalize()
+    assert (
+        "finalize_job"
+        not in ip.get_ipython()
+        .run_cell("VDK.job._plugin_registry.hook().__dict__")
+        .result
+    )
+
+    ip.get_ipython().run_cell("VDK.finalize()")
+
+    # check whether finalize_job is in the output after calling finalize()
+    assert (
+        "finalize_job"
+        in ip.get_ipython().run_cell("VDK.job._plugin_registry.hook().__dict__").result
+    )
+
+
+def test_get_initialized_job_input_multiple_times(ip):
+    ip.run_line_magic(magic_name="reload_VDK", line="")
+
+    ip.get_ipython().run_cell("job_input = VDK.get_initialized_job_input()")
+    first_result = ip.get_ipython().run_cell("job_input").result
+    ip.get_ipython().run_cell("job_input = VDK.get_initialized_job_input()")
+
+    assert first_result == ip.get_ipython().run_cell("job_input").result
+
+
+def test_get_initialized_job_input_after_finalize(ip):
+    ip.run_line_magic(magic_name="reload_VDK", line="")
+
+    ip.get_ipython().run_cell("job_input = VDK.get_initialized_job_input()")
+    ip.get_ipython().run_cell("VDK.finalize()")
+    ip.get_ipython().run_cell("job_input = VDK.get_initialized_job_input()")
+    ip.get_ipython().run_cell("VDK.finalize()")
+    ip.get_ipython().run_cell("job_input = VDK.get_initialized_job_input()")
+
+    assert ip.user_global_ns["job_input"] is not None
+    assert isinstance(ip.user_global_ns["job_input"], IJobInput)
+
+
+def test_call_finalize_multiple_times(ip):
+    ip.run_line_magic(magic_name="reload_VDK", line="")
+
+    ip.get_ipython().run_cell("VDK.finalize()")
+    ip.get_ipython().run_cell("job_input = VDK.get_initialized_job_input()")
+    ip.get_ipython().run_cell("VDK.finalize()")
+    # verifying that calling finalize multiple times won't produce any errors (the method will not fail)
```

Comment:
I think that's not enough verification. 

We want to 

A) Assert that the new job_input is different than the now finalized , right ? 
B) Assert that finalize has been called for both new and old job_input.

Reply:
It might be good idea to verify that the second VDK.finalize does nothing. We do not want to finalize the job twice.

Reply:
the test is about testing the first finalise(), not the second. The first one does nothing (even though there is no initialized job calling finalise will not end with an error).

Reply:
I mean this method test_call_finalize_multiple_times and the next method test_call_finalize_before_get_initialized_job_input are the same ?

Reply:
I changed them